### ISSUE TITLE
treemacs dired: unwanted indentation for directories

### ISSUE THREAD

**Issue by mcraveiro** [Alexander-Miller/treemacs#846]
Hi treemacs developers,

many thanks for an extremely useful mode, on which I rely daily. I have a small problem, which I am sure is related to my config [1], but I am not entirely sure how. When i enable ```treemacs-icons-dired```, I have some spacing on folders and a greater than (">"):

![Screenshot from 2021-09-17 15-38-26](https://user-images.githubusercontent.com/180654/133801735-0710319d-5306-4e0a-a9bb-c57fcef360a9.png)

I'd like files and folders to be lined up as per the screenshot in the main page [2]. Can you please tell me what I am doing wrong? My config options were sourced from the examples, but no doubt I managed to mess one of them up :-)

Many thanks for your time.

Marco

[1] https://github.com/mcraveiro/cunene/blob/master/cunene.org#treemacs
[2] https://github.com/Alexander-Miller/treemacs

**Comment by Alexander-Miller** [comment]
Pretty sure this is an example for why I don't like iconic fonts. Here's what I think is happening:

That iconic font you are using is not monospaced, so there is always some ugly misalignment between the icons and the text. To compensate you can add tabs to the icons, like the doom theme you are using did. Except the tabs are usually too large, so things end up looking ugly anyway. 

Try adding a mode-hook to change the `tab-width` in dired buffers, that should largely mitigate the problem.

**Comment by mcraveiro** [comment]
Hi @Alexander-Miller,

Thanks very much for the prompt response. I'm afraid I am not up-to-speed with regards to iconic fonts but I can change my font if that helps things. At present I am using ```Source Code Pro 9```, because it seems to provide me with the icons necessary to render the "powerline" git prompt in eshell [1]. So if I understood you correctly, I need to find a monospaced version for ```Source Code Pro```? Sorry, just so I can get my head around this issue.

Many thanks for your time.

[1] https://github.com/xuchunyang/eshell-git-prompt

**Comment by Alexander-Miller** [comment]
> So if I understood you correctly, I need to find a monospaced version for Source Code Pro?

Nope. Your normal text font has nothing to do with this. An iconic font is literally some font that contains a bunch of icons instead of/in addition to the usual letters and numbers and symbols.

The fonts used for icons are always the same: https://github.com/domtronn/all-the-icons.el#resource-fonts. I don't know if the doom-themes package you use depends on all-the-icons directly or simply uses the same fonts, but switching your text font won't help. Your options are:

- try reducing the tab width to see if it helps
- Try the official treemacs-all-the-icons theme instead of the doom version
- use treemacs' original png icons

**Comment by mcraveiro** [comment]
Thanks very much, it was exactly the ```tab-width```! Now, I am not sure how it is that my ```tab-width``` is getting set to 8 in dired, given that my config specifies 4, but that's neither here or there as the "correct" value appears to be 1! :-) Its all lined up correctly now:

![Screenshot from 2021-09-23 21-01-44](https://user-images.githubusercontent.com/180654/134575789-211952ed-9fcf-44da-9552-979b1a2b7c76.png)

For anyone facing this issue, here are some references that alude to the problem @Alexander-Miller explains above: [1], [2].

Many thanks for all your help.

[1] [emacs-doom-themes #293: tabs in dired](https://github.com/h
lissner/emacs-doom-themes/issues/293)
[2] [emacs-doom-themes: Fix #293](https://github.com/hlissner/emacs-doom-themes/commit/3571d462f0bcd7b87ab07e977090e61cb2c6a2f3)

**Comment by mcraveiro** [comment]
For good measure, here's the code I used to setup ```tab-width```:

```lisp
(defun cunene/setup-tab-width-treemacs-dired (&rest _)
  "Set `tab-width' to 1, so tab characters don't ruin formatting."
  (setq tab-width 1))

(add-hook 'treemacs-mode-hook #'cunene/setup-tab-width-treemacs-dired)
```
It is a copy and paste variation from [1].

[1] [emacs-doom-themes: Fix #293](https://github.com/hlissner/emacs-doom-themes/commit/3571d462f0bcd7b87ab07e977090e61cb2c6a2f3)

**Comment by mcraveiro** [comment]
Actually, that did not fix the issue it seems. A second approach seems to be more fruitful:

```lisp
(add-hook 'dired-mode-hook
          (lambda ()
            (setq-local tab-width 1)))
```
